templates/BUILD.bazel

```diff
@@ -18,26 +19,28 @@ genrule(
         SRC=templates
         OUT=templates-tarball
 
-        # skeleton template
-        mkdir -p $$OUT/skeleton
-        cp -rL $$SRC/skeleton/* $$OUT/skeleton/
-
-        # migrate template
-        mkdir -p $$OUT/migrate
-        cp -rL $$SRC/migrate/* $$OUT/migrate/
+        # templates in templates dir
+        for d in skeleton migrate; do
+            mkdir -p $$OUT/$$d
+            cp -rL $$SRC/$$d/* $$OUT/$$d/
+            cp $$SRC/default-gitignore $$OUT/$$d/.gitignore
+        done
```

Comment:
Have you considered having one `for` loop over `skeleton`, `migrate`, `quickstart-java` and `quickstart-scala` that copies the `.gitignore` file? Would that be cleaner?

Reply:
Good point. There is actually more in common to all cases than I at first noticed. I've merged the loops and only kept the special cases.

> *NOTE*: This is slightly changing the semantics of the template generation, in that with this changes, if a file exists under both `templates/quickstart-*` and the quickstart's original location, the quickstart will "win" (overwrite), whereas in the previous setup the templates one would win. This is not an issue right now as there is no common file.

We should probably work on moving all the templates to a single directory. I think my preference would be to have this build rule in `daml-assistant/BUILD.bazel` and all the templates (and only the templates) under `daml-assistant/templates`, so the script could just be
```
for d in templates/*; do
    mkdir -p $$OUT/$$d
    cp -rL $$SRC/templates/* $$OUT/$$d/
    cp $$SRC/default-gitignore $$OUT/$$d/.gitignore
done
```
and we don't need to remember to add to the Bazel script when we add a template. I'm not sure I have enough context to trust my preference though (e.g. I have no idea why the java and scala templates are elsewhere, or even why they're called quickstart-* rather than just java and scala).

Reply:
That preference sounds good to me.
@associahedron @leo-da Do you know why the `quickstart-*` templates are living in a different place than the others?

Reply:
The `quickstart-java` template lives where it does because the template source code gets spliced into docs by sphinx.